app/Command/Compile.hs

```diff
@@ -122,21 +126,45 @@ jsonErrors = Opts.switch $
      Opts.long "json-errors"
   <> Opts.help "Print errors to stderr as JSON"
 
-sourceMaps :: Opts.Parser Bool
-sourceMaps = Opts.switch $
-     Opts.long "source-maps"
-  <> Opts.help "Generate source maps"
-
-dumpCoreFn :: Opts.Parser Bool
-dumpCoreFn = Opts.switch $
-     Opts.long "dump-corefn"
-  <> Opts.help "Dump the (functional) core representation of the compiled code at output/*/corefn.json"
+codegenTargets :: Opts.Parser [P.CodegenTarget]
+codegenTargets = Opts.option targetParser $
+     Opts.short 'g'
+  <> Opts.long "codegen"
+  <> Opts.value [P.JS]
+  <> Opts.help
+      ( "Specifies comma-separated codegen targets to include. "
+      <> targetsMessage
+      <> " The default target is 'js', but if this option is used only the targets specified will be used."
+      )
+
+targets :: M.Map String P.CodegenTarget
+targets = M.fromList
+  [ ("js", P.JS)
+  , ("sourcemaps", P.JSSourceMap)
+  , ("corefn", P.CoreFn)
+  ]
+
+targetsMessage :: String
+targetsMessage = "Accepted codegen targets are '" <> intercalate "', '" (M.keys targets) <> "'."
+
+targetParser :: Opts.ReadM [P.CodegenTarget]
+targetParser =
+  Opts.str >>= \s ->
+    for (T.split (== ',') s)
```

Comment:
Do we want to be a bit more lenient here and strip the text after splitting?

Reply:
I'm not sure how you'd get spaces into that value anyway? Since `--codegen js, corefn` wouldn't parse as one thing.

Reply:
Oh... quotes maybe?